var/spack/repos/builtin/packages/m4/package.py

```diff
@@ -19,7 +19,9 @@ class M4(AutotoolsPackage):
     patch('pgi.patch', when='@1.4.17')
     # from: https://github.com/Homebrew/homebrew-core/blob/master/Formula/m4.rb
     # Patch credit to Jeremy Huddleston Sequoia <jeremyhu@apple.com>
-    patch('secure_snprintf.patch', when='platform_os = highsierra')
+    patch('secure_snprintf.patch', when='os = highsierra')
+    patch('secure_snprintf.patch', when='os = mojave')
+    patch('secure_snprintf.patch', when='os = catalina')
```

Comment:
@alalazo would something like this work?
```python
patch('secure_snprintf.patch', when='os=highsierra:')
```

Reply:
No, we didn't add any comparison semantic to the `os`.